addons/mail/static/src/views/form/form_controller.js

```diff
@@ -15,16 +15,27 @@ import { evalDomain } from "@web/views/utils";
 
 import { MailFormCompiler } from "./form_compiler";
 
-const { onMounted, onWillUnmount } = owl;
+const { onMounted, onWillDestroy, onWillUnmount } = owl;
 
 patch(FormController.prototype, "mail", {
     setup() {
         this._super();
         this.uiService = useService("ui");
         this.hasAttachmentViewerInArch = false;
+        this.chatter = undefined;
 
         if (this.env.services.messaging) {
             useModels();
+            this.env.services.messaging.modelManager.messagingCreatedPromise.then(() => {
+                if (owl.status(this) === "destroyed") {
+                    return;
+                }
+                const messaging = this.env.services.messaging.modelManager.messaging;
+                this.chatter = messaging.models['Chatter'].insert({ id: getChatterNextTemporaryId() });
+                if (owl.status(this) === "destroyed") {
+                    this.chatter.delete();
+                }
```

Comment:
Is it really possible for an OWL component to change flag "destroyed" during sequence of synchronous steps?

Reply:
Yes, `insert` is synchronous and it is calling `render` synchronously as well, and `render` does some synchronous stuff such as potentially destroying components that were in the process of existing but are no longer relevant with the new state, or something like that from what I can remember.

Reply:
`render -> makeRootFiber -> cancelFibers` probably

Reply:
Ok, so make sense to keep it.

FYI we have also something similar in Discuss models: let's say `record` exists, and we do an update that makes this `record` no longer exist. Following synchronous LOCs must guard `exists`.

Thankfully this happen quite rarely in current modelling, but we should be aware that this is something that will become more problematic in the long run. On a positive side, it makes computed field more attractive, because functions will have too many guards here and there if we keep using them.